remotespark/sparkkernelbase.py

```diff
@@ -54,16 +56,17 @@ def do_execute(self, code, silent, store_history=True, user_expressions=None, al
             self._abort_with_fatal_error(self._fatal_error)
 
         subcommand, flags, code_to_run = self._parse_user_command(code)
+        msg_printer = MessagePrinter.MessagePrinter()
```

Comment:
I don't understand why you're creating one MessagePrinter in SparkKernelBase and another in SparkController. Why wouldn't you be using the same one all the time?